aten/src/ATen/native/Activation.cpp

```diff
@@ -252,25 +262,6 @@ Tensor & celu_(Tensor & self, const Scalar& alpha) {
   return at::elu_(self, alpha, Scalar(1.0), Scalar(inv_alpha));
 }
 
-Tensor silu(const Tensor& self) {
-  Tensor result = at::empty({0}, self.options());
-  at::silu_out(result, self);
-  return result;
-}
-
-Tensor& silu_(Tensor& self) {
-  return at::silu_out(self, self);
-}
-
-Tensor& silu_out(const Tensor& self, Tensor& result) {
-  TORCH_CHECK(
-      result.dtype() == self.dtype(),
-      "Output Tensor should have the same type as in Input Tensor.")
```

Comment:
hmmmmm I wonder what's going on with this check

Reply:
This check is not needed, because it is already checked in build_unary_op that the input and output have the same type.
```
void TensorIteratorBase::build_unary_op(const Tensor& out, const Tensor& a) {
  build(TensorIteratorConfig()
      .set_check_mem_overlap(true)
      .add_output(out)
      .add_input(a)
      .cast_common_dtype_to_outputs(false)
      .enforce_safe_casting_to_output(false)
      .check_all_same_dtype(true));    <--------HERE
}
```
Older code shouldn't need this check either